cmd/terminal.go

```diff
@@ -416,6 +425,18 @@ func (c *TerminalCmd) pipeStdout(conn *websocket.Conn, w io.Writer) {
 			if _, err := w.Write(m.Body); err != nil {
 				break
 			}
+		} else if m.Header.Proto == ws.ProtoTypeShell && m.Header.MsgType == wsshell.MessageTypePingShell {
+			if healthcheckTimeout, ok := m.Header.Properties["timeout"].(int64); ok && healthcheckTimeout > 0 {
+				c.healthcheck <- int(healthcheckTimeout)
+			}
```

Comment:
if `! ok`, shouldn't we not reply with Pong?

Reply:
If the mender-connect disables the client-side timeout (e.g. not setting timeout), we don't do the healthcheck; it is not supported as of today (the timeout is hard-coded), but who knows in the future... in this case, the mender-cli still replies to "ping" requests with a pong, but does not set any client-side timeout. To be honest, the important thing is being sure the user (mender-cli) is still around, the other way around is a "nice to have" feature.

Interestingly enough, same question from Manuel: https://github.com/mendersoftware/gui/pull/1193#discussion_r553919019